tests/wpt_tests.cpp

```diff
@@ -0,0 +1,105 @@
+//#include <cstring>
+#include <filesystem>
+#include <iostream>
+//#include <map>
+//#include <memory>
+#include <set>
+#include <sstream>
+
+#include "gtest/gtest.h"
+#include "mimesniff.h"
+
+// This function copies your input onto a memory buffer that
+// has just the necessary size. This will entice tools to detect
+// an out-of-bound access.
+//template <class result_type = ada::url_aggregator>
+//ada::result<result_type> ada_parse(std::string_view view,
+//                                   const result_type *base = nullptr) {
+//  std::cout << "about to parse '" << view << "' [" << view.size() << " bytes]"
+//            << std::endl;
+//  std::unique_ptr<char[]> buffer(new char[view.size()]);
+//  memcpy(buffer.get(), view.data(), view.size());
+//  return ada::parse<result_type>(std::string_view(buffer.get(), view.size()),
+//                                 base);
+//}
+//
+//template ada::result<ada::url> ada_parse(std::string_view view,
+//                                         const ada::url *base);
+//template ada::result<ada::url_aggregator> ada_parse(
+//    std::string_view view, const ada::url_aggregator *base);
+
+#include "simdjson.h"
+
+using namespace simdjson;
+
+#ifndef WPT_DATA_DIR
+#define WPT_DATA_DIR "wpt/"
+#endif
+const char *GENERATED_MIME_TYPES_JSON = WPT_DATA_DIR "generated-mime-types.json";
+const char *MIME_TYPES_JSON = WPT_DATA_DIR "mime-types.json";
+
+//using Types = testing::Types<ada::mimesniff::parse_mime_type(), ada::url_aggregator>;
+//template <class T>
+//struct wpt_tests_typed : testing::Test {};
+//TYPED_TEST_SUITE(wpt_tests_typed, Types);
```

Comment:
Since we don't have 2 different types (in ada we have ada::url, and ada::url_aggregator), we can just remove these lines.